sdk/core/core-arm/src/lroPollStrategy.ts

```diff
@@ -358,9 +423,12 @@ class LocationLROPollStrategy extends LROPollStrategy {
     const initialResponse: HttpOperationResponse = this._pollState.initialResponse;
     const initialRequestMethod: HttpMethods = initialResponse.request.method;
     const statusCode: number = parsedResponse.status;
-    if (statusCode === 200 ||
-      (statusCode === 201 && (initialRequestMethod === "PUT" || initialRequestMethod === "PATCH")) ||
-      (statusCode === 204 && (initialRequestMethod === "DELETE" || initialRequestMethod === "POST"))) {
+    if (
```

Comment:
this looks less readable after styling...I think we should refactor if it's going to be this bad